src/vs/workbench/common/editor/editorGroup.ts

```diff
@@ -740,4 +729,26 @@ export class EditorGroup extends Disposable {
 			this.preview = this.editors[data.preview];
 		}
 	}
+
+	// {{SQL CARBON EDIT}}
+	async removeNonExitingEditor(): Promise<void> {
+		let n = 0;
+		while (n < this.editors.length) {
+			let editor = this.editors[n];
+			if (editor instanceof QueryInput && editor.matchInputInstanceType(FileEditorInput) && !editor.isDirty() && await editor.inputFileExists() === false && this.editors.length > 1) {
```

Comment:
Details on the if conditions here
1. editor instanceof QueryInput : only query editor (for later checks to work)
2. editor.matchInputInstanceType(FileEditorInput) : which has been created from fileEditorInput (not untitled input)
3. !editor.isDirty() : does not have changes
4. editor.inputFileExists() : actual file does not exist
5. editors.length > 1: opened with at least one more editor because single active editor scenario is handled by VS and if we do it here then welcome page doesn't show at reopen

Reply:
@anthonydresser  please take a look when you get a chance